app/controllers/form_controller.rb

```diff
@@ -109,11 +106,19 @@ def responses_for_page(page)
   end
 
   def find_resource
-    @lettings_log = current_user.lettings_logs.find_by(id: params[:id])
+    @log = if URI(request.referer).path.split("/")[1] == "sales-logs"
```

Comment:
Ouch! I don't think this is going to work reliably. Remember that referer is entirely optional. Can you think of a better way to do this?

Reply:
We'll definitely try to! I think this was the first attempt we came up with so definitely scope for improving this 😄